lib/worker.go

```diff
@@ -43,15 +42,18 @@ func NewWorker(cfg *WorkerConfig) (w Worker, err error) {
 // LocalWorker is an in-process implementation of worker
 // TODO - finish parallelism implementation
 type LocalWorker struct {
-	coord   WorkCoordinator
-	stop    chan bool
-	cfg     *WorkerConfig
-	fetcher *fetchbot.Fetcher
-	queue   *fetchbot.Queue
+	coord    WorkCoordinator
+	stop     chan bool
+	cfg      *WorkerConfig
+	fetchers []*fetchbot.Fetcher
+	queues   []*fetchbot.Queue
```

Comment:
One the one hand, I feel like it would be better for the some other component to create multiple LocalWorkers instead of leaving it to the worker itself to manage its own parallelism, but on the other hand, I know the Chrome Headless one will benefit from working this way because it can manage a single browser process and have a separate tab for each parallel implementation.

So I guess this makes sense. I just wish it didn’t. :P